src/Avalonia.Controls/Platform/ManagedDispatcherImpl.cs

```diff
@@ -99,18 +97,18 @@ public void RunLoop(CancellationToken token)
                 continue;
             }
 
-            TimeSpan? nextTimer;
+            long? nextTimer;
             lock (_lock)
             {
-                nextTimer = _nextTimer;
+                nextTimer = _nextTimerMs;
             }
 
             if (nextTimer != null)
             {
-                var waitFor = nextTimer.Value - _clock.Elapsed;
-                if (waitFor.TotalMilliseconds < 1)
+                long waitFor = nextTimer.Value - _clock.ElapsedMilliseconds;
+                if (waitFor < 1 || waitFor > int.MaxValue)
```

Comment:
While *very* unlikely (read: impossible with the current render timer) to have a timer running for more than `int.MaxValue` milliseconds, you might want to clamp `waitFor` to `int.MaxValue` and wait for that duration instead of skipping the wait, to avoid a potential never-paused situation again.